Script loader

L3nn0x opened the pull request:
Loads and execute the lua scripts. Only creates npcs in the ecs for testing purposes.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dev-osrose/osirose-new/133)
<!-- Reviewable:end -->

RavenX8 committed:
Added packets for npcs

L3nn0x committed:
Create script_loader.h

L3nn0x committed:
Create script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Updated to only load .lua files

Commit:
Added script path

Commit:
Updated script loader

L3nn0x committed:
Update script_loader.h

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update entitysystem.cpp

L3nn0x committed:
Update script_loader.h

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.h

L3nn0x committed:
Update script_loader.h

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.h

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update entitysystem.cpp

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update script_loader.cpp

L3nn0x committed:
Update entitysystem.cpp

L3nn0x committed:
Update config.h

L3nn0x committed:
Update main.cpp

L3nn0x committed:
Update cmapserver.h

L3nn0x committed:
Update config.h

L3nn0x committed:
Update cmapserver.h

L3nn0x committed:
Update cmapserver.cpp

L3nn0x committed:
Create root.lua

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update entitysystem.cpp

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update cmapserver.h

L3nn0x committed:
Update cmapserver.h

L3nn0x committed:
Update cmapserver.cpp

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update entitysystem.cpp

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update cmapclient.h

L3nn0x committed:
Update chatsystem.cpp

L3nn0x committed:
Update cmapserver.h

L3nn0x committed:
Update cmapserver.cpp

L3nn0x committed:
Update cmapserver.h

L3nn0x committed:
Update cmapclient.h

L3nn0x committed:
Update entitysystem.h

L3nn0x committed:
Update entitysystem.cpp

Commit:
Solve compilation issues

Commit:
Added needed stuff for npcs

Commit:
Send npcs to clients on arrival on map

Commit:
Corrected a silly bug

Commit:
Missing include

Commit:
Corrected bugs

Commit:
Added filtering by map id

L3nn0x committed:
Added all npcs in cities

RavenX8 commented on src/map/src/main.cpp:
```diff
@@ -169,7 +169,7 @@ int main(int argc, char* argv[]) {
                                  config.database().database, config.database().host, config.database().port));
 
-    CMapServer clientServer;
+    CMapServer clientServer(false, 2); // FIXME: use the config file option (hardcoded to zant for now)
```
Please resolve this. This needs both a config file and command line option.

RavenX8 commented on scripts/npcs/cities/JPT01.lua:
```diff
@@ -1,30 +1,30 @@
 
 --[[ NPC SPAWN LIST
-npc(<npc_lua_file>, <npc_id>, <map_id>, <x_pos>, <y_pos>, <z_pos>, <angle>);
+npc(<npc_lua_ile>, <npc_id>, <map_id>, <x_pos>, <y_pos>, <z_pos>, <angle>);
```
please capture the updated version of the scripts in trunk.

Commit:
Corrected small typo

Commit:
Added command line args

Commit:
Updated

Commit:
Update

Commit:
Merge trunk

Commit:
Added more stuff

RavenX8 committed:
Fixed curl compile error for visual studio if you aren't using ninja.
Fixed issue where NPCs wouldn't talk to you (srv_mousecmd.cpp was sending the wrong field).

RavenX8 committed:
Updated all of the NPC scripts to have the dialog IDs

RavenX8 committed:
Fixed Character select issue where the server would log you into the wrong character.
